geometric/neb.py

```diff
@@ -292,19 +293,49 @@ def ComputeEnergyGradient(self, cyc=None, result=None):
         """Compute energies and gradients for each structure."""
         # This is the parallel point.
         wq = getWorkQueue()
-        if wq is None:
-            for i in range(len(self)):
-                if result:
-                    self.Structures[i].ComputeEnergyGradient(result=result[i])
-                else:
-                    self.Structures[i].ComputeEnergyGradient()
-        else:  # If work queue is available, handle jobs with the work queue.
+        if wq:  # If work queue is available, handle jobs with the work queue.
             for i in range(len(self)):
                 self.Structures[i].QueueEnergyGradient()
             wq_wait(wq, print_time=600)
             for i in range(len(self)):
                 self.Structures[i].GetEnergyGradient()
-        if cyc is not None:
+        elif self.params.bigchem:
+            # If BigChem is available, it will be used to carry the single-point calculations.
+            from qcio import Molecule as qcio_Molecule, ProgramInput
+            from bigchem import compute, group
+            elems = self.Structures[0].M.elem
+            molecules = []
+            
+            # Creating a list with qcio Molecule objects and submitting calculations.
+            for Structure in self.Structures:
+                molecules.append(qcio_Molecule(symbols=elems, geometry=Structure.cartesians.reshape(-1,3)))
+
+            outputs = group(compute.s(self.engine.__class__.__name__.lower(),
+                            ProgramInput(molecule=qcio_M, calctype="gradient",
+                                         model={"method":self.engine.method, "basis": self.engine.basis},
+                                         extras={"order":i}),
+                            ) for i, qcio_M in enumerate(molecules)).apply_async()
+
+            # Getting the records
+            records = outputs.get()
+            
+            # Passing the results to chain.ComputeEnergyGradient()
+            for i in range(len(self)):
+                assert records[i].input_data.extras["order"] == i
+                result = {"energy": records[i].results.energy, "gradient": np.array(records[i].results.gradient).ravel()}
+                self.Structures[i].ComputeEnergyGradient(result=result)
+
+            # Deleting the records
+            outputs.forget()
```

Comment:
You can do this instead if you want to make it simpler:

```python
from bigchem import parallel_hessian

pi = ProgramInput(molecule=qcio_m, calctype='hessian', model={...}, extras={...})
fr = parallel_hessian(self.engine.basis, pi).delay()
output = fr.get()
fr.forget()
hessian = output.results.hessian
```

No worries if you want to keep your own implementation. You can see my implementation of this [here](https://github.com/mtzgroup/bigchem/blob/2850cf7a8ed15993f30a4a442d575b4d339a56c5/bigchem/algos.py#L20-L53).

Reply:
This part is for NEB, which won't need the Hessian calculation.

Reply:
Ah, I see! I mistakenly thought it was a hessian :)